### PR TITLE
Update IndexedDB API

### PR DESCRIPTION
**Pull request by falsandtru**
Fixes https://github.com/Microsoft/TypeScript/issues/14872, https://github.com/Microsoft/TypeScript/issues/15483

### CODE REVIEW

File: baselines/dom.generated.d.ts [hunk 1]
```diff
@@ -7112,7 +7112,7 @@ interface IDBRequestEventMap {
 }
 
 interface IDBRequest extends EventTarget {
-    readonly error: DOMError;
+    readonly error: DOMException | DOMError;
```
Comment (SaschaNaz) [inline baselines/dom.generated.d.ts]: DOMError is deprecated, isn't DOMException enough?

Comment (falsandtru) [inline baselines/dom.generated.d.ts]: I cannot decide it. @mhegazy ?

Comment (SaschaNaz) [inline baselines/dom.generated.d.ts]: Structurally DOMException is a subtype of DOMError so it shouldn't cause any compatibility issue.

Comment (falsandtru) [inline baselines/dom.generated.d.ts]: These two have an important difference.

```
> new DOMError('') instanceof Error
false
> new DOMException('') instanceof Error
true
```

Comment (SaschaNaz) [inline baselines/dom.generated.d.ts]: Right but that's 100% browser dependent behavior and TS type does not affect it.

Comment (falsandtru) [inline baselines/dom.generated.d.ts]: Then we lose to know this difference through TS type. Type names are important info too.

Comment (falsandtru) [inline baselines/dom.generated.d.ts]: @mhegazy What do you think?

**Comment by falsandtru** [comment]
@mhegazy

File: baselines/dom.generated.d.ts (hunk 1 above)
Comment (mhegazy) [inline baselines/dom.generated.d.ts]: I would say `DOMException`. from usage perspective `DOMException` has the same props as `DOMError`, so it is not going to break existing code. but it allows for users using new browsers to use the new APIs.

Comment (falsandtru) [inline baselines/dom.generated.d.ts]: Done, but travis seems to have a problem. Please restart the build after it is fixed.

**Pull request merged**